libr/crypto/p/crypto_rol.c

```diff
@@ -25,7 +27,9 @@ static bool rol_init(struct rol_state *const state, const ut8 *key, int keylen)
 static void rol_crypt(struct rol_state *const state, const ut8 *inbuf, ut8 *outbuf, int buflen) {
 	int i;
 	for (i = 0; i < buflen; i++) {
-		outbuf[i] = inbuf[i] << state->key[i%state->key_size];
+		ut8 count = state->key[i % state->key_size] & 7;
+		ut8 inByte = inbuf[i];
+		outbuf[i] = (inByte << count) | (inByte >> ((8 - count) & 7));
```

Comment:
In case someone wonders, the `(8 - count) & 7` serves to protect from shifting with 8; because shifting with more or equal to the width of the type is formally incorrect in C (undefined behavior).

Do not worry about performance here, bitwise `and` is cheap, and the compiler will probably be smart enough to optimize out the operation anyway.